src/Tasks/system.design/stronglytypedresourcebuilder.cs

```diff
@@ -497,34 +538,18 @@ private static bool DefineResourceFetchingProperty(String propertyName, String r
                 HasSet = false
             };
 
-            Type type = data.Type;
-            if (type == null)
+            string typeName = data.TypeName;
+            if (typeName == null)
             {
                 return false;
             }
 
-            if (type == typeof(MemoryStream))
+            if (typeName == typeof(MemoryStream).FullName)
             {
-                type = typeof(UnmanagedMemoryStream);
+                typeName = typeof(UnmanagedMemoryStream).FullName;
             }
 
-            // Ensure type is internalally visible.  This is necessary to ensure
-            // users can access classes via a base type.  Imagine a class like
-            // Image or Stream as a internalally available base class, then an 
-            // internal type like MyBitmap or __UnmanagedMemoryStream as an 
-            // internal implementation for that base class.  For internalally 
-            // available strongly typed resource classes, we must return the 
-            // internal type.  For simplicity, we'll do that for internal strongly 
-            // typed resource classes as well.  Ideally we'd also like to check
-            // for interfaces like IList, but I don't know how to do that without
-            // special casing collection interfaces & ignoring serialization 
-            // interfaces or IDisposable.
-            while (!type.IsPublic)
-            {
-                type = type.BaseType;
-            }
```

Comment:
It's not clear to me why this can be deleted. Won't it break existing uses?

Reply:
It had to be deleted because I'm no longer operating on types. I am dealing with the full names of types as strings, because MSBuild might not have access to the types (such as Bitmap) that the user is placing in their resx file.

Reply:
I see that. I think we would have to leave the prior behavior on full-framework msbuild with more conditional compilation. :(

Reply:
yeah, I'm afraid so :(

Reply:
@Dreddy-work

Reply:
Just my 2 cents, but I feel it would be really useful to have implementations of `System.Resources.IResourceReader` and `System.Resources.IResourceWriter` available to MSBuild and cross-platform (hopefully wrapped as a .netstandard package so it can fall through to the full framework or use a custom implementation etc.)
I believe that EF6 will need this as part of its porting.